redbot/cogs/warnings/warnings.py

```diff
@@ -301,19 +346,43 @@ def __init__(self, bot: Red):
 
         await warning_points_add_check(self.config, ctx, user, current_point_count)
         try:
+            dm = await self.config.guild(ctx.guild).toggle_dm()
+            if dm == True:
+                em = discord.Embed(
+                    title=_("Warning from {user}").format(user=ctx.author),
+                    description=reason_type["description"],
+                )
+                em.add_field(name=_("Points"), value=str(reason_type["points"]))
+                await user.send(
+                    _("You have received a warning in {guild_name}.").format(
+                        guild_name=ctx.guild.name
+                    ),
+                    embed=em,
+                )
+                await ctx.send(_("User {user} has been warned in their DM's.").format(user=user))
+        except discord.HTTPException:
+            await ctx.send(
+                "The warning couldn't be sent to the user because the user has DM's from server members turned off."
+            )
+        toggle_channel = await self.config.guild(ctx.guild).toggle_channel()
+        if toggle_channel == True:
             em = discord.Embed(
                 title=_("Warning from {user}").format(user=ctx.author),
                 description=reason_type["description"],
             )
             em.add_field(name=_("Points"), value=str(reason_type["points"]))
-            await user.send(
-                _("You have received a warning in {guild_name}.").format(
-                    guild_name=ctx.guild.name
-                ),
-                embed=em,
+            warn_channel = await self.config.guild(ctx.guild).warn_channel()
+            if warn_channel == None:
```

Comment:
```suggestion
            if warn_channel is None:
```
You should just use identity check, because it's singleton.